### DISCUSSION TITLE
Is there a way to retrieve ANSI-encoded test results from TestReport objects?

### DISCUSSION THREAD

**Discussion by jeffwright13** [pytest-dev/pytest#9831] [Q&A] [answered]
I know TestReport has a `longreprtext` attribute that gives the traceback info for failed tests. Is there similar attribute/method that gives me the same ANSI-encoded string as it would be sent to the terminal?

**Comment by jeffwright13** [comment, marked as answer]
Question was addressed in part in [this StackExchange thread](https://stackoverflow.com/questions/71846269/algorithm-for-extracting-first-and-last-lines-from-sectionalized-output-file/71869524#71869524).

**Comment by nicoddemus** [comment]
Just to mention that you can probably also create a `TerminalWriter` and call `report.toterminal(terminal_writer)`. 👍
